dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeLiteConnector.java

```diff
@@ -50,72 +55,54 @@ public SnowflakeLiteConnector() {
         SnowflakeTaskUtil.withFilter(format, usageView, usageZip, usageFilter, header);
-    ImmutableList<Task<?>> tasks = getSqlTasks(inputSource, header, format, schemaTask, usageTask);
-    out.addAll(tasks);
+    return getSqlTasks(inputSource, header, format, schemaTask, usageTask);
   }
 
   @Override
-  public final void addTasksTo(
-      @Nonnull List<? super Task<?>> out, @Nonnull ConnectorArguments arguments) {
+  public final void addTasksTo(List<? super Task<?>> out, ConnectorArguments arguments) {
     out.add(new DumpMetadataTask(arguments, FORMAT_NAME));
     out.add(new FormatTask(FORMAT_NAME));
 
-    boolean INJECT_IS_FAULT = arguments.isTestFlag('A');
-    final String IS = INJECT_IS_FAULT ? "__NONEXISTENT__" : "INFORMATION_SCHEMA";
-    final String AU = "SNOWFLAKE.ACCOUNT_USAGE";
-    final String AU_WHERE = " WHERE DELETED IS NULL";
+    out.addAll(
+        sqlTasksWithInfoSchemaFallback(
+            DatabasesFormat.Header.class,
+            "SELECT database_name, database_owner FROM %1$s.DATABASES%2$s",
+            DatabasesFormat.IS_ZIP_ENTRY_NAME,
+            DatabasesFormat.AU_ZIP_ENTRY_NAME));
 
-    addSqlTasksWithInfoSchemaFallback(
-        out,
-        DatabasesFormat.Header.class,
-        "SELECT database_name, database_owner FROM %1$s.DATABASES%2$s",
-        DatabasesFormat.IS_ZIP_ENTRY_NAME,
-        IS,
-        DatabasesFormat.AU_ZIP_ENTRY_NAME,
-        AU,
-        AU_WHERE);
+    out.addAll(
+        sqlTasksWithInfoSchemaFallback(
+            SchemataFormat.Header.class,
+            "SELECT catalog_name, schema_name FROM %1$s.SCHEMATA%2$s",
+            SchemataFormat.IS_ZIP_ENTRY_NAME,
+            SchemataFormat.AU_ZIP_ENTRY_NAME));
 
-    addSqlTasksWithInfoSchemaFallback(
-        out,
-        SchemataFormat.Header.class,
-        "SELECT catalog_name, schema_name FROM %1$s.SCHEMATA%2$s",
-        SchemataFormat.IS_ZIP_ENTRY_NAME,
-        IS,
-        SchemataFormat.AU_ZIP_ENTRY_NAME,
-        AU,
-        AU_WHERE);
-
-    addSqlTasksWithInfoSchemaFallback(
-        out,
-        TablesFormat.Header.class,
-        "SELECT table_catalog, table_schema, table_name, table_type, row_count, bytes,"
-            + " clustering_key FROM %1$s.TABLES%2$s",
-        TablesFormat.IS_ZIP_ENTRY_NAME,
-        IS,
-        TablesFormat.AU_ZIP_ENTRY_NAME,
-        AU,
-        AU_WHERE);
+    out.addAll(
+        sqlTasksWithInfoSchemaFallback(
+            TablesFormat.Header.class,
+            "SELECT table_catalog, table_schema, table_name, table_type, row_count, bytes,"
+                + " clustering_key FROM %1$s.TABLES%2$s",
+            TablesFormat.IS_ZIP_ENTRY_NAME,
+            TablesFormat.AU_ZIP_ENTRY_NAME));
 
     if (arguments.isAssessment()) {
       for (AssessmentQuery item : planner.generateAssessmentQueries()) {
-        String query = String.format(item.formatString, AU, /* an empty WHERE clause */ "");
+        String usageSchema = "SNOWFLAKE.ACCOUNT_USAGE";
```

Comment:
this string is the same as in line 64, so could be moved as field as well. Also it creates the same string in a loop, so worth to moved out of the loop.

Reply:
The repetition is bigger than the two occurrences, because it's also in the metadata connector. I'm planning to remove all of these variables, but it's a slightly bigger change so I'd leave it to the next PR. In this PR I can bring back the parameter, which would remove one repeated occurrence. WDYT?

The string literal is a constant, so an object is not created, this only sets a reference to the constant. Using the minimal necessary scope is an advantage here.